acos_client/v30/axapi_http.py

```diff
@@ -64,6 +64,16 @@ def request(self, method, api_url, params={}, headers=None,
         LOG.debug("axapi_http: %s url = %s", method, api_url)
         LOG.debug("axapi_http: params = %s", json.dumps(params, indent=4))
 
+        # Update params with **kwargs for currently unsupported configuration
+        # of objects
+        formatted_kwargs = dict([(k.replace('_', '-'), v) for k, v in kwargs.iteritems()])
+        param_keys = params.keys()
+        if params != {}:
```

Comment:
Correct. This is not ideal, but unfortunately the json is always formatted with that top level key. For example something like:

```json
{
    "virtual-port": {
        "port": 80,
        "protocol": "tcp"
    }
}
```

So if params was empty, I would not know how to format `**kwargs` as we are currently placing them in the top level `dict` or `params`.

Totally open to suggestions if you can see a way around this.